bootloader/wscript

```diff
@@ -244,9 +244,16 @@ def detect_arch(ctx):
             arch = '64bit'
             ctx.env.DEST_CPU = 'arm64'
     else:
-        # PyInstaller uses the result of platform.architecture() to determine the bits and this is testing the pointer
-        # size (via module struct). We do the same here.
-        arch = "%sbit" % (8 * check_sizeof_pointer(ctx))
+        if system == "Windows" and ctx.env.CC_NAME == "msvc":
+            # With MSVC, detection based on the compiler doesn't make much sense since the compiler is selected
+            # based on what waf (wrongly) thinks the build platform is. Set target platform to that of the current
+            # Python environment.
+            ctx.env.DEST_CPU = os.environ.get("PROCESSOR_ARCHITECTURE", ctx.env.DEST_CPU)
+            arch = "32bit" if os.environ.get("PROCESSOR_ARCHITECTURE", ctx.env.DEST_CPU) == "x86" else "64bit"
```

Comment:
There's no need to query `PROCESSOR_ARCHITECTURE` again - checking `ctx.env.DEST_CPU` should suffice.